vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -250,4 +248,56 @@ public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
     }
+    
+    /**
+     * Retrieves a list of IVSImages for pagination based on the provided parameters.
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param category - The category of the images to filter by.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List<VSImage> getPaginatedImagesBySearchTerm(int pageNo, ImageCategory category, String searchTerm,
+            String sortedBy, String order) {    	
+    	Sort sortingParameters = getSortingParameters(sortedBy, order);
+        pageNo = validatePageNumber(pageNo, category);
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
+        String likeSearchTerm = "%" + searchTerm + "%";
+        Page<VSImage> results;
+        if(category!=null) {
+            results = imageRepo.findByCategoryAndFilenameLikeOrNameLikeOrDescriptionLike(
+                    sortByRequestedField, category, likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        }
+        else {
+            results = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(sortByRequestedField,
+                    likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        }
+        return results.getContent();
+    }
+      
+    /**
+     * Method to return the total pages sufficient to display all images
+     * with respect to the search text
+     * 
+     * @param searchTerm - This is the search string which is being searched.
+     * 
+     * @return totalPages - this is the total pages for the search results
+     *  
+     */
+    @Override
+    public long getTotalPagesOnSearchText(String searchTerm) {
```

Comment:
Do you really need these methods? the page object in you get when querying the repository has `totalElements` and `totalPages`.